### ISSUE TITLE
Use os.ErrNotExist instead of custom errors?

### ISSUE THREAD

**Issue by fridolin-koch** [jteeuwen/go-bindata#114]
Hi,

I'm using https://github.com/elazarl/go-bindata-assetfs to serve assets via `http.FileServer`. When requesting non existent files I get `500 Internal Server Error` instead of `404 Not Found`. After some research I found the reason here: https://golang.org/src/net/http/fs.go?s=13011:13069#L425

So maybe it would be better to use `os.ErrNotExist` instead of the custom errors in `AssetDir()`, `Asset()` and `AssetInfo()`?

See also: https://github.com/fridolin-koch/go-bindata/commit/87513f25697d396dd17951eca11985a3d973c035

Best Regards
Frido

**Comment by shurcooL** [comment]
> See also: fridolin-koch@87513f2

It might be better to use `os.PathError` to keep additional information about the path that wasn't found, but still have it so `os.IsNotExist(err)` returns true:

```Go
return nil, &os.PathError{Op: "open", Path: path, Err: os.ErrNotExist}
```

Source: https://github.com/shurcooL/vfsgen/blob/ba39b2120b3137bc1867bfc242bae267bcf2f9bb/generator.go#L319.

That way instead of:

```
file does not exist
```

You get:

```
open /foo/bar: file does not exist
```

**Comment by fridolin-koch** [comment]
I agree, thats better. I changed it https://github.com/fridolin-koch/go-bindata/commit/41cb39b77a818b580e54e85751cac239c9cfea59

**Comment by markwallsgrove** [comment]
@fridolin-koch can you raise a PR for this? Fixes https://github.com/elazarl/go-bindata-assetfs/pull/26#issuecomment-168428267

**Comment by fridolin-koch** [comment]
Done https://github.com/jteeuwen/go-bindata/pull/123

**Comment by xiaods** [comment]
need it +1

**Comment by TamalSaha** [comment]
We would like to see this. Can this be added?
